ios/Exponent/Kernel/Services/Notifications/EXRemoteNotificationManager.m

```diff
@@ -56,18 +56,38 @@ - (void)registerForRemoteNotifications
     // don't register, because the detached app may not be built with APNS entitlements,
     // and in that case this method would actually be bad to call. (not just a no-op.)
     DDLogWarn(@"Expo Remote Notification services won't work in an ExpoKit app because Expo cannot manage your APNS certificates.");
+    
+    // registering for remote notifications triggers [self registerAPNSToken] and fires memorized promises resolution
```

Comment:
I think it would be helpful to clearly say that registering for remote notifications happens in a different code path:

> when the app is eligible to register for remote notifications, we eventually call [self registerAPNSToken] and fulfill the pending promises, but in this code path we never register for notifications so we need to reject the pending promises manually